[FLINK-24090][docs] Added Troubleshooting section with ignoring in-fl… (#17126)
apache/flink: docs/content/docs/ops/state/unaligned_checkpoints.md

from subtask 0 of the stateful operator. Upon restore, we send that copy to all of the
operators. Therefore it might happen that an operator will get the state with changes applied for a
record that it will soon consume from its checkpointed channels.

## Troubleshooting

### Corrupted in-flight data
{{< hint warning >}}

> Actions described below are a last resort due to they lead to data loss. Which means they should be applied only at extreme scenarios.

Reviewer: ```suggestion
Actions described below are a last resort as they will lead to data loss.
```

{{< /hint >}}

> In case of the in-flight data corrupted or by another reason when the job should be restored without in-flight data, 

Reviewer: ```suggestion
In case of the in-flight data corrupted or by another reason when the job should be restored without the in-flight data, 
```

it is possible to use  [recover-without-channel-state.checkpoint-id]({{< ref "docs/deployment/config" >}}#execution-checkpointing-recover-without-channel-state-checkpoint) property.

> This property requires the checkpoint id of the checkpoint for which in-flight data should be ignored.

Reviewer: ```suggestion
This property requires to specify a checkpoint id for which in-flight data will be ignored.
```

> Extremely important to keep this property with default value in order to avoid accidental triggering of ignoring in-flight data.

Reviewer: ```suggestion
Do not set this property, unless a corruption inside the persisted in-flight data has lead to an otherwise unrecoverable situation.
```
